### ISSUE TITLE
ColumnStore closed connection

### ISSUE THREAD

**Issue by IlianYotov** [MariaDB/mariadb-connector-nodejs#67]
Hello,

I'm have a problem with a query using "with" that run against columnstore and innodb table. It throws ER_SOCKET_UNEXPECTED_CLOSE. 

Here is a example query that I want to try
```
WITH 
  users AS (
SELECT user_id,
       activated_at,
       state,
       language
  FROM tutorial.playbook_users
),

  events AS (
SELECT user_id,
       event_name,
       occurred_at,
       device,
       location
  FROM tutorial.playbook_events
)

SELECT x.cohort AS "Date",
       MAX(x.period_age) OVER (PARTITION BY x.cohort) AS cutoff_age,
       x.period_age,
       x.tally,
       x.tally/MAX(x.tally) OVER (PARTITION BY x.cohort)::FLOAT AS retention_rate,
       MAX(x.tally) OVER (PARTITION BY x.cohort) AS "New Users"
  FROM (
SELECT DATE_TRUNC('week',u.activated_at) AS cohort,
       FLOOR(EXTRACT('day' FROM e.occurred_at - u.activated_at)/7) AS period_age,
       COUNT(DISTINCT u.user_id) AS tally
  FROM users u
  LEFT JOIN events e
    ON e.user_id = u.user_id
 WHERE u.activated_at IS NOT NULL
   AND u.activated_at >= '2014-05-01'
 GROUP BY 1,2
       ) x
 ORDER BY 1,2,3
```

where playbook_users is a innodb table and playbook_events are columnstore table.
The same query executed from MySQL Workbench works fine. Am I missing something?

**Comment by IlianYotov** [comment]
Just to clarify that the query is rewrited for mariadb ( e.g. EXTRACT/DATE_TRUNC are converted for mariadb funcs )

**Comment by IlianYotov** [comment]
It looks that the problem is not with nodejs driver because I could reproduce it with java driver/mcsmysql client, but works fine with workbench...

**Comment by drrtuy** [comment]
Could you send me relevant DDL to try to reproduce the error?

**Comment by yotov** [comment]
I found what causes the problem. It appears when there is no default schema so Mariadb closes the connection and there was a segmentation error in logs ( e.g. kernel: mysqld[15354]: segfault at 0 ip 00007fe398ed58e9 sp 00007fe35447fa90 error 4 in libcalmysql.so.1.0.0[7fe398e30000+128000]
 ). 

The problem occurs on Columnstore 1.2.3 and it seems to be fixed in 1.2.5

**Comment by drrtuy** [comment]
GTK. I wonder what fixes this b/c I couldn't recall the relevant change
from the top of my head.

вт, 13 авг. 2019 г., 14:02 Ilian Yotov <notifications@github.com>:
